File: Web/src/app/models/report.filter.model.ts
export class ReportFilterModel {
    AcademicYearId: string;
    DivisionId: string;
    DistrictId: string;
    SectorId: string;
    JobRoleId: string;
    ClassId: string;
    VTPId: string;
    PageIndex: number;
    PageSize: number;
    IgnoreCriteria: boolean;

    constructor() {
        this.AcademicYearId = '';
        this.DivisionId = '';
        this.DistrictId = '';
        this.SectorId = '';
        this.JobRoleId = '';
        this.ClassId = '';
        this.VTPId = '';
        this.PageIndex = 1;
        this.PageSize = 10000;
    }
}

File: Services/Igmite.Lighthouse.Models/Reports/ReportFilterModel.cs
using System;
using System.Runtime.Serialization;

namespace Igmite.Lighthouse.Models
{
    public class ReportFilterModel
    {
        [DataMember]
        public virtual string UserId { get; set; }

        [DataMember]
        public virtual Guid? AcademicYearId { get; set; }

        [DataMember]
        public virtual string MonthId { get; set; }

        [DataMember]
        public virtual Guid? DivisionId { get; set; }

        [DataMember]
        public virtual string DistrictId { get; set; }

        [DataMember]
        public virtual Guid? SectorId { get; set; }

        [DataMember]
        public virtual Guid? JobRoleId { get; set; }

        [DataMember]
        public virtual Guid? VTPId { get; set; }

        [DataMember]
        public virtual Guid? VTId { get; set; }

        [DataMember]
        public virtual Guid? VCId { get; set; }

        [DataMember]
        public virtual Guid? HMId { get; set; }

        [DataMember]
        public virtual Guid? SchoolId { get; set; }

        [DataMember]
        public virtual Guid? ClassId { get; set; }

        [DataMember]
        public virtual Guid? SectionId { get; set; }

        [DataMember]
        public virtual Guid? GenderId { get; set; }

        [DataMember]
        public virtual string SchoolManagementId { get; set; }

        [DataMember]
        public virtual DateTime? ReportDate { get; set; }

        [DataMember]
        public virtual DateTime? FromDate { get; set; }

        [DataMember]
        public virtual DateTime? ToDate { get; set; }

        [DataMember]
        public virtual string Name { get; set; }

        [DataMember]
        public virtual int PageIndex { get; set; }

        [DataMember]
        public virtual int PageSize { get; set; }
    }
}